adapters/sinatra/lib/opentelemetry/adapters/sinatra/adapter.rb

```diff
@@ -0,0 +1,67 @@
+# frozen_string_literal: true
+
+# Copyright 2019 OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+require 'sinatra'
+require 'opentelemetry'
+
+require_relative 'extensions/tracer_extension'
+
+module OpenTelemetry
+  module Adapters
+    module Sinatra
+      class Adapter
+        class << self
+          attr_reader :config,
+                      :propagator
+
+          def install(config = {})
+            @config = config
+            @propagator = tracer_factory.http_text_format
+
+            new.install
+          end
+
+          def registry
+            @registry ||= {}
+          end
+
+          def tracer
+            @tracer ||= OpenTelemetry.tracer_factory.tracer(
+              Sinatra.name,
+              Sinatra.version
+            )
+          end
+
+          private
+
+          def tracer_factory
+            OpenTelemetry.tracer_factory
+          end
+        end
+
+        def install
+          register_tracer_extension
+        end
+
+        private
+
+        def register_tracer_extension
```

Comment:
We are likely going to need an `InstrumentationRegistry` that is part of the OpenTelemetry::API to ensure that instrumentation is installed only once. Is the point of this code to fake that we have a registry so that it can be easily changed later?

If so, I'm wondering if it wouldn't be better for us to just have a class level `installed?` flag for now, and update it to use the actual registry later?

Reply:
I've switched to the `installed?` flag.